### PR TITLE
[Preprocess] Set default trainable to false @open sesame 04/22 12:39

### LINKED ISSUE #1129: Check preprocess layer
1. Add on app_context to create layer
2. Add preprocess layer in example
3. Check preprocess layer in backpropagation

### PR DESCRIPTION
**Pull request by lhs8928**
[Unittest] Added doxygen tag
    
    1. Added missing doxygen tag
    2. remove unused function parameter
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

[Preprocess] Set default trainable to false

**Changes Proposed**
- Set default trainable to `false` for preprocess layer
- Register preprocess layer to the app context
- Add validate test

resolves #1129

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/layers/preprocess_flip_layer.h [hunk 1]
```diff
@@ -34,7 +34,9 @@ class PreprocessFlipLayer : public Layer {
   template <typename... Args>
   PreprocessFlipLayer(Args... args) :
     Layer(args...),
-    flipdirection(FlipDirection::horizontal_and_vertical) {}
+    flipdirection(FlipDirection::horizontal_and_vertical) {
+    trainable = false;
```
Comment (kparichay) [inline nntrainer/layers/preprocess_flip_layer.h:38]: good catch

File: test/unittest/unittest_nntrainer_models.cpp [hunk 2]
```diff
@@ -90,7 +92,7 @@ class NodeWatcher {
    */
   NodeWatcher(const NodeType &node) : node(node) {
     unsigned int num_weights = node.layer->getNumWeights();
-    if (node.layer->getType() != nntrainer::InputLayer::type)
+    if (node.layer->getType() != nntrainer::InputLayer::type && node.layer->getType() != nntrainer::PreprocessFlipLayer::type && node.layer->getType() != nntrainer::PreprocessTranslateLayer::type)
```
Comment (zhoonit) [inline test/unittest/unittest_nntrainer_models.cpp:95]: Could you check clang-format?

Comment (lhs8928) [reply test/unittest/unittest_nntrainer_models.cpp]: Done

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**